stl/inc/format

```diff
@@ -1129,10 +1129,15 @@ public:
     template <class _Ty>
     _NODISCARD constexpr unsigned long long operator()(const _Ty _Value) const {
         if constexpr (is_integral_v<_Ty>) {
+            bool _Positive;
             if constexpr (is_signed_v<_Ty>) {
-                if (_Value < 0) {
-                    _THROW(format_error("Negative width."));
-                }
+                _Positive = _Value > 0;
+            } else {
+                _Positive = _Value != 0;
+            }

```

Comment:
Do we need `if constexpr` logic anymore? I believe that `const bool _Positive = _Value > 0;` would be readable and efficient for both signed and unsigned types.

Reply:
You are correct, but sadly `bool` is accepted here and `bool > 0` triggers C4804. I'll make this typeswitch detect exactly `bool` and comment it for maximum clarity.